test/extensions/common/wasm/test_data/test_context_cpp.cc

```diff
@@ -58,24 +85,29 @@ bool TestRootContext::onDone() {
 
 // Null VM fails on nullptr.
 void TestRootContext::onTick() {
-  if (envoy_resolve_dns(nullptr, 1, &dns_token_) != WasmResult::InvalidMemoryAccess) {
-    logInfo("resolve_dns should report invalid memory access");
-  }
-  if (envoy_resolve_dns("example.com", sizeof("example.com") - 1, nullptr) !=
-      WasmResult::InvalidMemoryAccess) {
-    logInfo("resolve_dns should report invalid memory access");
+  if (test_ == "resolve dns") {
```

Comment:
The anti-pattern I'm suggesting to not do here is thread around the code a bunch of logic conditioning on strings from the caller. Please create meaningful options struct or other state variables/parameters to reflect what you want `setup` and the test fixture to do.

Reply:
agreed cc @PiotrSikora

Reply:
done